synapse/cortex.py

```diff
@@ -2075,13 +2079,51 @@ async def _delStormCmd(self, name):
 
         await self.fire('core:cmd:change', cmd=name, act='del')
 
-    async def addStormPkg(self, pkgdef):
+    async def addStormPkg(self, pkgdef, verify=False):
         '''
         Add the given storm package to the cortex.
 
         This will store the package for future use.
         '''
         # do validation before nexs...
+        if verify:
+            pkgcopy = s_msgpack.deepcopy(pkgdef)
+            codesign = pkgcopy.pop('codesign', None)
+            if codesign is None:
+                mesg = 'Storm package is not signed!'
+                raise s_exc.BadPkgDef(mesg=mesg)
+
+            certbyts = codesign.get('cert')
+            if certbyts is None:
+                mesg = 'Storm package has no certificate!'
+                raise s_exc.BadPkgDef(mesg=mesg)
+
+            signbyts = codesign.get('sign')
+            if signbyts is None:
+                mesg = 'Storm package has no signature!'
+                raise s_exc.BadPkgDef(mesg=mesg)
+
+            cert = self.certdir.loadCertByts(certbyts)
```

Comment:
This can also raise OpenSSL errors from bad certificates:
```
cli> storm $lib.pkg.add(({'codesign': {'cert': 'foo', 'sign': 'bar'}}), verify=$lib.true)
Executing query at 2022/12/22 15:43:28.913
ERROR: ('Error', {'efile': '_util.py', 'eline': 57, 'mesg': "[('PEM routines', '', 'no start line')]"})
```

Reply:
certdir.loadCertBytes needs to raise a synerr varient. I will patch that.

Reply:
This now raises a synerr subclass